src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -16,6 +17,14 @@
           Condition="'$(RestorePackages)'=='true' AND '$(IsTestProject)' == 'true' AND '$(TestRuntimeProjectJson)' != ''">
     <Exec Command="$(DnuRestoreCommand) &quot;$(TestRuntimeProjectJson)&quot;" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
   </Target>
+  
+  <Target Name="ValidateTestRuntimeDependencies"
+          AfterTargets="RestoreTestRuntimePackage"
```

Comment:
Thanks! It looks like it might be straightforward to add that semaphore. Looking deeper into it, I'm seeing that a run from the root *sometimes* skips `RestoreTestRuntimePackage` because `RestorePackages` is false, but for some test projects it runs anyway:

    System.Globalization\tests\System.Globalization.Tests.csproj
    System.Composition.Convention\tests\System.Composition.Convention.Tests.csproj
    System.Composition\tests\System.Composition.Tests.csproj
    System.Diagnostics.Tools\tests\System.Diagnostics.Tools.Tests.csproj
    System.IO\tests\System.IO.Tests.csproj
    System.Runtime\tests\System.Runtime.Tests.csproj
    System.Text.Encodings.Web\tests\System.Text.Encodings.Web.Tests.csproj
    System.Runtime.InteropServices\tests\System.Runtime.InteropServices.Tests.csproj
    System.Runtime.Extensions\tests\System.Runtime.Extensions.Tests.csproj

I'm trying to find a reason for this. I noticed though that for both toolsruntime and testruntime, `BatchRestorePackages` restores the project.json and then later it's restored again, so there might be other strangeness going on. Should I look further into this or just set up a semaphore for now?